GWAS.pm

```diff
@@ -0,0 +1,484 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 CONTACT
+
+ Ensembl <http://www.ensembl.org/info/about/contact/index.html>
+
+=cut
+
+=head1 NAME
+
+ GWAS
+
+=head1 SYNOPSIS
+
+ mv GWAS.pm ~/.vep/Plugins
+ ./vep -i variations.vcf --plugin GWAS,file=/FULL_PATH_TO/gwas_catalog_v1.0.2-associations_e107_r2022-09-14.tsv
+ ./vep -i variations.vcf --plugin GWAS,type=sstate,file=/FULL_PATH_TO/17463246-GCST000028-EFO_0001360.h.tsv.sorted.gz
+
+=head1 DESCRIPTION
+
+ A VEP plugin that retrieves GWAS catalog data given the file. 
+ 
+ This plugin supports both the curated data that is found in the download section of the GWAS catalog website and the 
+ summary statistics file. By default the plugin assumes the file provided is the curated file but you can pass "type=sstate" 
+ to say you want to annotate with a summary statistics file.
+
+ Please cite the following publication alongside the VEP if you use this resource:
+ https://pubmed.ncbi.nlm.nih.gov/30445434/
+
+ Pre-requisites:
+
+ For curated GWAS catalog file -
+ GWAS files can be downloaded from - https://www.ebi.ac.uk/gwas/api/search/downloads/alternative
+ When run for the first time the plugin will create a processed file that have genomic locations and indexed and put it under 
+ the --dir location determined by Ensembl VEP. It may take >1 hour to create the processed file depending on the file size. But subsequent 
+ runs will be faster as the plugin will be using the already generated processed file.
+ 
+ For summary statistics file -
+ The plugin can process the harmonised version of the summary statistics file. Which can be downloaded from the FTP site - 
+ http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics
+ 
+ They are under directory with related to their specific GCST id. For example -
+ http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST000001-GCST001000/GCST000028/harmonised/17463246-GCST000028-EFO_0001360.h.tsv.gz
+ 
+ Please keep the filename format as it is because filename is parsed to get information.
+ 
+ Follow the following steps -
+ 1. filter out lines that don't have genomic locations -
+ tail -n +2 17463246-GCST000028-EFO_0001360.h.tsv | awk '$3 != "NA" && $4 != "NA" {print}' > 17463246-GCST000028-EFO_0001360.h.tsv.filtered
+
+ replace $3 and $4 with column location of the hm_chrom and hm_pos
+ 
+ 2. sort the file. Also keep the header and put # in front - 
+ (head -n 1 17463246-GCST000028-EFO_0001360.h.tsv && sort -k3,3 -k4,4n 17463246-GCST000028-EFO_0001360.h.tsv.filtered) > 17463246-GCST000028-EFO_0001360.h.tsv.sorted
+ sed -i '1 s/^/#/' 17463246-GCST000028-EFO_0001360.h.tsv.sorted
+
+ 3. zip the file -
+ bgzip -c 17463246-GCST000028-EFO_0001360.h.tsv.sorted > 17463246-GCST000028-EFO_0001360.h.tsv.sorted.gz
+
+ 4. index the file -
+ tabix -s 3 -b 4 -e 4 -c "#" -f 17463246-GCST000028-EFO_0001360.h.tsv.sorted.gz
```

Comment:
I think these steps are a little bit complicated, can we store statistics file based on header + have a check to make sure we have "hm_chrom" + "hm_pos" in file header?